src/util/simplify_expr_pointer.cpp

```diff
@@ -417,40 +418,78 @@ bool simplify_exprt::simplify_pointer_offset(exprt &expr)
   return true;
 }
 
-bool simplify_exprt::simplify_inequality_address_of(exprt &expr)
+/// Remove any typecasts around the expression
+static exprt strip_typecasts(const exprt &expr)
 {
-  assert(expr.type().id()==ID_bool);
-  assert(expr.operands().size()==2);
-  assert(expr.id()==ID_equal || expr.id()==ID_notequal);
+  exprt result = expr;
```

Comment:
No need to construct a temporary `exprt`. Just keep a pointer. And then return a `const exprt &`.